### PR TITLE
Cleanup: simplifies conditions

### PR DESCRIPTION
**Pull request by lkwg82**
### Purpose

simplifies code (negations)

### Testing

run unit tests

### CODE REVIEW

File: lib/model/model.go [hunk 1]
```diff
@@ -1396,20 +1396,20 @@ func (m *Model) CurrentFolderFile(folder string, file string) (protocol.FileInfo
 	m.fmut.RLock()
 	fs, ok := m.folderFiles[folder]
 	m.fmut.RUnlock()
-	if !ok {
-		return protocol.FileInfo{}, false
+	if ok {
+		return fs.Get(protocol.LocalDeviceID, file)
 	}
-	return fs.Get(protocol.LocalDeviceID, file)
+	return protocol.FileInfo{}, false
```
Comment (AudriusButkevicius) [inline lib/model/model.go]: This makes it worse, if not ok, early return is easier to process mentally

File: lib/model/model.go [hunk 2]
```diff
@@ -1396,20 +1396,20 @@ func (m *Model) CurrentFolderFile(folder string, file string) (protocol.FileInfo
...
 	m.fmut.RLock()
 	fs, ok := m.folderFiles[folder]
 	m.fmut.RUnlock()
-	if !ok {
-		return protocol.FileInfo{}, false
+	if ok {
+		return fs.GetGlobal(file)
 	}
-	return fs.GetGlobal(file)
+	return protocol.FileInfo{}, false
```
Comment (AudriusButkevicius) [inline lib/model/model.go]: Same here.

File: lib/model/model.go [hunk 3]
```diff
@@ -2282,13 +2282,12 @@ func (m *Model) CurrentSequence(folder string) (int64, bool) {
...
-		// specified folder name from the REST interface.
-		return 0, false
+	if ok {
+		return fs.Sequence(protocol.LocalDeviceID), true
 	}
-
-	return fs.Sequence(protocol.LocalDeviceID), true
+	// specified folder name from the REST interface.
+	// The folder might not exist, since this can be called with a user
+	return 0, false
```
Comment (AudriusButkevicius) [inline lib/model/model.go]: And here.

**Comment by AudriusButkevicius** [comment]
I think a bunch of these makes things worse, as nit ok, early return is easier to process mentally

**Comment by calmh** [comment]
Yes. The idiom is to have the failure/error/unlikely cases checked, handled, and early returned from. The outer level of the function/method is the success case and should end in return nil etc. The `if !ok` and `if err != nil` things are so ingrained that they require no mental effort to process.

**Comment by lkwg82** [comment]
yeah ... I would like to challenge that.

File: lib/model/rwfolder.go [hunk 4]
```diff
@@ -1495,6 +1491,8 @@ func (f *sendReceiveFolder) finisherRoutine(ignores *ignore.Matcher, in <-chan *
 				blockStats["copyOriginShifted"] += state.copyOriginShifted
 				blockStats["copyElsewhere"] += state.copyTotal - state.copyOrigin
 				blockStatsMut.Unlock()
+			} else {
+				f.newError("finisher", state.file.Name, err)
```
Comment (AudriusButkevicius) [inline lib/model/rwfolder.go]: This is not even equivalent to what it was, as the error returned by performFinish no longer causes f.newError as it's within the non-error if case now.

Comment (lkwg82) [reply lib/model/rwfolder.go]: yes, I just recognized and reverted

**Comment by AudriusButkevicius** [comment]
Challange in what way? This is how most go code is written.

**Comment by lkwg82** [comment]
I just read your PR on crypto and thought some things could get cleared beforehand. 

> Challange in what way? This is how most go code is written.

Beside the idiomatic way, the code has a high complexity. I tried to remove some pieces.

> This is how most go code is written.

Sry, 95% of my time I write java. :)

File: lib/model/model.go [hunk 5]
```diff
@@ -2557,11 +2557,10 @@ func (m *Model) checkDeviceFolderConnectedLocked(device protocol.DeviceID, folde
...
 	}
 
-	if !m.folderDevices.has(device, folder) {
-		return errors.New("folder is not shared with device")
+	if m.folderDevices.has(device, folder) {
+		return nil
 	}
-
-	return nil
+	return errors.New("folder is not shared with device")
```
Comment (AudriusButkevicius) [inline lib/model/model.go]: I'd also revert this case, early return with an error makes sense.

**Comment by AudriusButkevicius** [comment]
@st-review merge

lib/model: Cleanup conditions

**Comment by st-review** [comment]
@AudriusButkevicius: Build status is `failure` -- refusing to merge.

**Comment by AudriusButkevicius** [comment]
@st-review retest

**Comment by st-review** [comment]
I'm sorry, @AudriusButkevicius. I'm afraid I don't know what you mean. I know how to `don't`, `lgtm`, `merge`, `prevent`, `rebuild`, `squash`, `stop`.

**Comment by AudriusButkevicius** [comment]
@st-review rebuild

**Comment by AudriusButkevicius** [comment]
@st-review merge

lib/model: Cleanup conditions

**Comment by st-review** [comment]
@AudriusButkevicius: Build status is `pending`. I'll wait until it goes green and then merge!

**Comment by AudriusButkevicius** [comment]
@st-review merge

lib/model: Cleanup conditions

**Comment by st-review** [comment]
@AudriusButkevicius: There's already a merge pending for this PR.

**Comment by AudriusButkevicius** [comment]
@st-review don't

**Comment by st-review** [comment]
@AudriusButkevicius: Preventing merge for the time being. Push a new revision to reset!

**Comment by st-review** [comment]
@AudriusButkevicius: Build status is `failure` -- refusing to merge.

**Comment by AudriusButkevicius** [comment]
@st-review merge

lib/model: Cleanup conditions

**Comment by st-review** [comment]
@AudriusButkevicius: Build status is `failure` -- refusing to merge.

**Comment by AudriusButkevicius** [comment]
@lkwg82 the bot is nuts, can you make a noop commit or something?

**Comment by AudriusButkevicius** [comment]
@st-review merge

lib/model: Cleanup conditions

**Comment by st-review** [comment]
@AudriusButkevicius: Build status is `pending`. I'll wait until it goes green and then merge!

**Comment by calmh** [comment]
You're getting everything built twice by using a branch on syncthing proper. Don't do that, use a fork.

**Comment by st-review** [comment]
:ok_hand: Merged as 9988044bbe637c50db59053672f8ae74df10e433. Thanks, @lkwg82!

**Pull request closed without merging**